### ISSUE TITLE
Backup command returns "django.db.utils.OperationalError: no such table: core_user " and fails

### ISSUE THREAD

**Issue by piperoc** [papermerge/papermerge-core#247]
This is a fantastic project. Thank you so much,

I installed with Portainer in minutes, and started using right away.

I am running the following versions:
Backend version: 2.1.8
Frontend version: 2.1.2

I tried to perform a backup following the instructions at [https://docs.papermerge.io/backup_restore.html](https://docs.papermerge.io/backup_restore.html ) but as soon as I run:

```shell
docker run papermerge/papermerge backup /place/backup/file/here/
```

it gives me the following output and fails:

```shell 
/venv/lib/python3.10/site-packages/requests/__init__.py:109: RequestsDependencyWarning: urllib3 (1.26.13) or chardet(None)/charset_normalizer (3.0.1) doesn't match a supported version!
  warnings.warn(Traceback (most recent call last):
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
  File "/venv/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 477, in execute
    return Database.Cursor.execute(self, query, params) 
sqlite3.OperationalError: no such table: core_user

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/manage.py", line 22, in <module> main()
  File "/app/manage.py", line 18, in main execute_from_command_line(sys.argv)
  File "/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line    utility.execute()
  File "/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 440, in execute    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/venv/lib/python3.10/site-packages/django/core/management/base.py", line 414, in run_from_argv self.execute(*args, **cmd_options)
  File "/venv/lib/python3.10/site-packages/django/core/management/base.py", line 460, in execute output = self.handle(*args, *options)
  File "/app/papermerge/core/management/commands/backup.py", line 41, in handle  backup_data(file_path=file_path)
  File "/app/papermerge/core/backup_restore/backup.py", line 227, in backup_data dict_data = dump_data_as_dict()
  File "/app/papermerge/core/backup_restore/backup.py", line 211, in dump_data_as_dict  result_dict['users'] = UserSerializer(User.objects, many=True).data
  File "/venv/lib/python3.10/site-packages/rest_framework/serializers.py", line 768, in data ret = super().data
  File "/venv/lib/python3.10/site-packages/rest_framework/serializers.py", line 253, in data self._data = elf.to_representation(self.instance)
  File "/venv/lib/python3.10/site-packages/rest_framework/serializers.py", line 686, in to_representation  return [
  File "/venv/lib/python3.10/site-packages/django/db/models/query.py", line 320, in __iter__    self._fetch_all()
  File "/venv/lib/python3.10/site-packages/django/db/models/query.py", line 1507, in _fetch_all   self._result_cache = list(self._iterable_class(self))
  File "/venv/lib/python3.10/site-packages/django/db/models/query.py", line 57, in __iter__   results = compiler.execute_sql(
  File "/venv/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1361, in execute_sql    cursor.execute(sql, params)
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 67, in execute   return self._execute_with_wrappers(
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers   return executor(sql, params, many, context)
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute  with self.db.wrap_database_errors:
  File "/venv/lib/python3.10/site-packages/django/db/utils.py", line 91, in __exit__   raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 89, in _execute   return self.cursor.execute(sql, params)
  File "/venv/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 477, in execute   return Database.Cursor.execute(self, query, params) django.db.utils.OperationalError: no such table: core_user
```

Thanks again for doing this.

**Comment by ciur** [comment]
This means that DB migrations did not run. Enter the application container and run:

$ ./manage.py migrate

**Comment by piperoc** [comment]
Thank you for replying so quickly.

I ran 

```shell
$ ./manage.py migrate
```

and got

```
/venv/lib/python3.10/site-packages/requests/__init__.py:109: RequestsDependencyWarning: urllib3 (1.26.13) or chardet (None)/charset_normalizer (3.0.1) doesn't match a supported version!
  warnings.warn(
Operations to perform:
  Apply all migrations: auth, authtoken, contenttypes, core, dynamic_preferences, dynamic_preferences_users, knox, sessions, sites
Running migrations:
  No migrations to apply.
  Your models in app(s): 'core' have changes that are not yet reflected in a migration, and so won't be applied.
  Run 'manage.py makemigrations' to make new migrations, and then re-run 'manage.py migrate' to apply them.
```

Therefore, I followed those instructions and got a positive result:

```shell
root@e05e990fae19:/app# ./manage.py makemigrations 
/venv/lib/python3.10/site-packages/requests/__init__.py:109: RequestsDependencyWarning: urllib3 (1.26.13) or chardet (None)/charset_normalizer (3.0.1) doesn't match a supported version!
  warnings.warn(
Migrations for 'core':
  papermerge/core/migrations/0010_alter_basetreenode_user_alter_tag_user.py
    - Alter field user on basetreenode
    - Alter field user on tag
root@e05e990fae19:/app# ./manage.py migrate
/venv/lib/python3.10/site-packages/requests/__init__.py:109: RequestsDependencyWarning: urllib3 (1.26.13) or chardet (None)/charset_normalizer (3.0.1) doesn't match a supported version!
  warnings.warn(
Operations to perform:
  Apply all migrations: auth, authtoken, contenttypes, core, dynamic_preferences, dynamic_preferences_users, knox, sessions, sites
Running migrations:
  Applying core.0010_alter_basetreenode_user_alter_tag_user... OK
```


After that, I assumed I  would be able to run the backup command. 

Instead, after running the same command 

```shell
placehold
```

I received the very same error

```shell
/venv/lib/python3.10/site-packages/requests/__init__.py:109: RequestsDependencyWarning: urllib3 (1.26.13) or chardet (None)/charset_normalizer (3.0.1) doesn't match a supported version!
  warnings.warn(Traceback (most recent call last):
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 89, in _execute  return self.cursor.execute(sql, params)
  File "/venv/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 477, in execute return Database.Cursor.execute(self, query, params)
sqlite3.OperationalError: no such table: core_user

The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/app/manage.py", line 22, in <module> main()
  File "/app/manage.py", line 18, in main execute_from_command_line(sys.argv)
  File "/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line    utility.execute()
  File "/venv/lib/python3.10/site-packages/django/core/management/__init__.py", line 440, in execute   self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/venv/lib/python3.10/site-packages/django/core/management/base.py", line 414, in run_from_argv self.execute(*args, **cmd_options)
  File "/venv/lib/python3.10/site-packages/django/core/management/base.py", line 460, in execute output = self.handle(*args, **options)
  File "/app/papermerge/core/management/commands/backup.py", line 41, in handle backup_data(file_path=file_path)
  File "/app/papermerge/core/backup_restore/backup.py", line 227, in backup_data dict_data = dump_data_as_dict()
  File "/app/papermerge/core/backup_restore/backup.py", line 211, in dump_data_as_dict result_dict['users'] = UserSerializer(User.objects, many=True).data
  File "/venv/lib/python3.10/site-packages/rest_framework/serializers.py", line 768, in data ret = super().data
  File "/venv/lib/python3.10/site-packages/rest_framework/serializers.py", line 253, in data self._data = self.to_representation(self.instance)
  File "/venv/lib/python3.10/site-packages/rest_framework/serializers.py", line 686, in to_representation  return [
  File "/venv/lib/python3.10/site-packages/django/db/models/query.py", line 320, in __iter__  self._fetch_all()
  File "/venv/lib/python3.10/site-packages/django/db/models/query.py", line 1507, in _fetch_all self._result_cache = list(self._iterable_class(self))
  File "/venv/lib/python3.10/site-packages/django/db/models/query.py", line 57, in __iter__ results = compiler.execute_sql(
  File "/venv/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1361, in execute_sql cursor.execute(sql, params)
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 67, in execute  return self._execute_with_wrappers(
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers return executor(sql, params, many, context)
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute with self.db.wrap_database_errors:
  File "/venv/lib/python3.10/site-packages/django/db/utils.py", line 91, in __exit__  raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/venv/lib/python3.10/site-packages/django/db/backends/utils.py", line 89, in _execute  return self.cursor.execute(sql, params)
  File "/venv/lib/python3.10/site-packages/django/db/backends/sqlite3/base.py", line 477, in execute  return Database.Cursor.execute(self, query, params)
django.db.utils.OperationalError: no such table: core_user
```

I am going to start looking into the warnings like `RequestsDependencyWarning` and then the rest of the call stacks, but I figured I'd give you this information in the meanwhile.

Also question: why would I need to run a "migration" process on a new installation? I might have missed that in the documentation.

Thank you again and I will share my findings while investigating too.

**Comment by piperoc** [comment]
While researching the dependency warning (which is a common Django warning with a lot of insights -- I do not know Django so I learned a few thing) it occurred to me to do a test on the backup process.

The documentation link that I mentioned explains 2 methods of backup:

1.  create a backup from the container by calling `manage.py backup`
2.  create a backup from the host by calling `docker run...`

I used the second because I want to save the backup files in a safe place outside docker as a disaster recovery measure.

However, I actually went into the backend container and I ran:

```shell
$ ./manage.py backup /app/media/
```

and that worked with no errors (still gave the dependency warning, _which means that the Django warning is not relevant to the backup performance_). And I was able to get the backup file in the volume, by going into `/var/lib/docker/volumes/papermerge_media_root/_data` on my host.

This way I can basically perform backups and restores through the docker volume while figuring our the original command.

I figured this might help someone who is in the same situation and equally oblivious of Django things.
